### ISSUE TITLE
REMINDER: The over-fitting issue on master branch

### ISSUE THREAD

**Issue by 1115712590** [Thinklab-SJTU/ThinkMatch#38]
This is a reminder for our users that we have noticed that **over-fitting sometimes occurs during training after we switched our code to pygmtools**. 

If you are also facing this issue, please switch from the ``master`` branch to [v0.1.0](https://github.com/Thinklab-SJTU/ThinkMatch/releases/tag/0.1.0). 

We are working to resolve it. We hope there will be a fix very soon.

**Comment by 1115712590** [comment]
We have resovled this issue. 

The occurance of the over-fitting is caused by **manual_seed**. We intend to **fix test data** to prevent test variance by setting an identical seed before evaluation in each epoch. The seed will also influence the training data. **We reset the seed to make it relative to the number of epoch, and reload the dataloader for training after each evaluation** to eliminate the influence.

Please refer to [my recent commit](https://github.com/Thinklab-SJTU/ThinkMatch/commit/336baac61f88079baaff242ac46ecde3238e2ae4), and the over-fitting issue on ``master`` branch shall be fixed now. If you have any further questions, feel free to open new issues or reopen this issue.
